### ISSUE TITLE
Backward compatibility error in PythonIdentifier validation regex

### ISSUE THREAD

**Issue by zopyx** [zopefoundation/zope.configuration#48]
See 

https://community.plone.org/t/transmogrifier-transmogrifier-registerconfig-with-dotted-name-as-name-parameter/8522

Short version: PythonIdentifier().validate("foo.bar") pass validation in former zope.configuration versions but no longer in the latest release.

**Comment by jamadden** [comment]
Yes, that's deliberate (https://github.com/zopefoundation/zope.configuration/pull/29). `DottedName` would be the field to use if you want to allow dots.
